src/hotspot/share/opto/bytecodeInfo.cpp

```diff
@@ -63,7 +63,8 @@ InlineTree::InlineTree(Compile* c,
     _caller_jvms->set_bci(caller_jvms->bci());
     assert(!caller_jvms->should_reexecute(), "there should be no reexecute bytecode with inlining");
   }
-  assert(_caller_jvms->same_calls_as(caller_jvms), "consistent JVMS");
+  assert(_caller_jvms == NULL
```

Comment:
Please, initialize `_caller_jvms` to `NULL` (on line 47), switch the null check to `caller_jvms` (on line 60), and  move the assert under the null check.

Reply:
I'm sorry, I've been looking at this for a while, and I can't figure out what "move the assert under the null check" means. There are three asserts here, and two of them are under the `if (caller_jvms != NULL)` check.

Reply:
> Please, initialize `_caller_jvms` to `NULL` (on line 47), switch the null check to `caller_jvms` (on line 60), a

Done.

Reply:
I'm talking specifically about the assert you are adjusting:
```
assert(_caller_jvms->same_calls_as(caller_jvms), "consistent JVMS");
```
It can be left as is once it is under `caller_jvms != NULL`, since there's no need in `_caller_jvms == NULL` anymore.

Reply:
Got it, done. Thanks.